scripts/common.py

```diff
@@ -97,6 +119,20 @@ def decode_container(container):
         items[i] = items[i][:items[i].find('_')] + ' ' + items[i][items[i].find('_')+1:]
     return ' '.join(items)
 
+def decode_container_as_dict(container):
+    """Decodes a container, i.e. the description of a location of a metadata table
+    for CCPP. Currently, there are three possibilities with different numbers of
+    input arguments: module, module+typedef, module+scheme+subroutine. Return
+    a dictionary with possible keys MODULE, TYPE, SCHEME, SUBROUTINE."""
+    items = container.split(' ')
+    if not len(items) in [1, 2, 3]:
+        raise Exception("decode_container not implemented for {0} items".format(len(items)))
+    itemsdict = {}
+    for i in xrange(len(items)):
```

Comment:
Noted. The CCPP prebuild scripts are only compatible with Python 2.7.x - something that has been criticized rightfully by others. Because we will move to capgen in the near (?) future, I am not making any attempts to make the CCPP prebuild scripts Python-3 compatible at this time.

Reply:
Right. Note that capgen is not yet python 3 compatible (but I keep picking at it when I have time).